3-With-Intune MAM/IntuneClient/active-directory-xamarin-intune.Android/IntuneMAMConnector.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using activedirectoryxamarinintune;
+using Microsoft.Identity.Client;
+using Microsoft.Intune.Mam.Client.App;
+using Microsoft.Intune.Mam.Policy;
+
+namespace active_directory_xamarin_intune.Droid
+{
+    public class IntuneMAMConnector : IIntuneMAMConnector
+    {
+        /// <summary>
+        /// Perform registration with MAM
+        /// </summary>
+        /// <param name="exProtection"></param>
+        /// <returns></returns>
+        public async Task DoMAMRegister(IntuneAppProtectionPolicyRequiredException exProtection)
+        {
+            // reset the registered event
+            PCAWrapper.MAMRegsiteredEvent.Reset();
+
+            // Invoke compliance API on a different thread
+            await Task.Run(() =>
+                                {
+                                    IMAMComplianceManager mgr = MAMComponents.Get<IMAMComplianceManager>();
+                                    mgr.RemediateCompliance(exProtection.Upn, exProtection.AccountUserId, exProtection.TenantId, exProtection.AuthorityUrl, false);
+                                }).ConfigureAwait(false);
+
+
+            // wait till the registration completes
+            // Note: This is a sample app for MSAL.NET. Scenarios such as what if enrollment fails or user chooses not to enroll will be as
```

Comment:
Instaed of this comment, can we provide a simplistic way of handlign errors? For example you can display a message ?

Reply:
There are three main scenarios:
1. User does not complete at all. In that case, there is no error and app just waits. 
2. User completes successfully and this should work.
3. An error occurs during the process. In that scenario, callback will be notified. But the connector will still wait. And there can be other stages between begging and successful registration. To handle that, it will require synchronization of the results between the waiting connector and the callback along with the event. Displaying message will make the app harder to understand. And it will be of little use to the developer as that app's business requirement may be different. Being developer's app, the cost to develop is too high compared to the benefit. My 2cents.